src/sawtoothMain/kotlin/lumberjack/sawtooth/component/PropertyComponent.kt

```diff
@@ -0,0 +1,18 @@
+package lumberjack.sawtooth.component
+
+import lumberjack.sawtooth.event.LogEvent
+import lumberjack.sawtooth.event.PropertyKey
+
+open class PropertyComponent<T : Any>(val property: PropertyKey<T>) : PatternComponent {
```

Comment:
My approach here was to allow a component to set up formatting when initialised, rather than when utilised, as to save on processing; this way, any necessary tuning can be done ahead of time, and just called once when needed.

I've changed this in my next commit to not have properties use this feature by default, but the convention now allows for subclasses to access a list of modifiers that are passed in on instantiation